src/main/java/com/softserve/repository/impl/LessonRepositoryImpl.java

```diff
@@ -259,4 +310,51 @@ public Integer updateLinkToMeeting(Lesson lesson) {
 
         return sessionFactory.getCurrentSession().createQuery(criteriaUpdate).executeUpdate();
     }
+
+    /**
+     * Method is used to update grouped lessons
+     * @param updatedLesson grouped lesson that needs to be updated
+     * @return updated Lesson
+     */
+    @Override
+    public Lesson updateGrouped(Lesson updatedLesson) {
+        log.info("Entered updateGroup({})", updatedLesson);
+        Lesson oldLesson = findById(updatedLesson.getId())
+                .orElseThrow(() -> new EntityNotFoundException(Lesson.class, "id", updatedLesson.getId().toString()));
```

Comment:
It'd be better to search old lesson on service side, and send already updated data to repo. So that in repo should be 1 method - 1 query